Implemented SelectMany with resultSelector.

meatGUY opened the pull request.

StefH commented on Src/System.Linq.Dynamic/ExpressionParser.cs:
```diff
@@ -799,12 +799,31 @@ Expression ParseIntegerLiteral()
             string text = _token.text;
+            string qualifier = null;
+            char last = text[text.Length - 1];
             if (text[0] != '-')
```
Negative values like `-5L` or `-5UL` are not parsed correctly.

StefH commented:
**Explicit integer qualifiers** : Negative values like `-5L` or `-5UL` are not parsed correctly.

test code:

```c#
var valuesL = new[] { -1L, -2L, -3L }.AsQueryable();
var resultValuesL = new[] { -2L, -3L }.AsQueryable();

//Act
var resultL = valuesL.Where("it <= -2L").ToDynamicArray<long>();

//Assert
Assert.Equal(resultL, resultValuesL.ToArray());
```

StefH commented:
I think there is a small error in your code, see my change here https://github.com/StefH/System.Linq.Dynamic.Core/issues/10 and check this and comment.

meatGUY commented:
Hi @StefH. Thank you for your findings. I have fixed the issue and also added an additional sanity check for (-2U) example and also extended 'in' operator to start parse from unary not primary expressions (because negative numbers are parsed through unary expressions).

StefH commented:
@meatGUY ; OK, this seems to work. I've copied your changes to my project. Can you maybe (if you have time) checkout my version and verify my issues 10 and 11 ?

BTW : I've noticed that parsing negative float or double does also not work very well. I'll take a look at that also.

StefH commented:
@meatGUY : I've also fixed the (negative) parsing from float and double ; see my issue 12.

However, for some reason the following does not work? (parsing 1.99 as double)

`var result2 = values.Where("it > 1.99").ToDynamicArray<double>();`

StefH commented:
Issue with parsing the `1.99` was related to my locale settings from windows. So I've changed from parsing code to
```c#
if (float.TryParse(text.Substring(0, text.Length - 1), NumberStyles.Float, CultureInfo.InvariantCulture, out f))
```
